### PR TITLE
Feat: ORV2-1930 Create Company Button on Empty Search Results

### PR DESCRIPTION
**Pull request by erikataot**
# Description
- Updated the IDIR company search results to display a friendly `Create Company` button when needed

image

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1174-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1174-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1174-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1174-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1174-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://onroutebc-1174-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1174-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1174-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1174-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1174-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/common/components/table/NoRecordsFound.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const NoRecordsFound = memo(() => {
     <div className="no-records-found">
       <div className="no-records-found__section no-records-found__section--img">
         <img
-          src="No_Data_Graphic.svg"
+          src="/No_Data_Graphic.svg"
```
Comment (erikataot) [inline frontend/src/common/components/table/NoRecordsFound.tsx:10]: this wasn't coming through properly for me...but how was it working for everyone else up to now? weird

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx [hunk 2]
```diff
@@ -153,8 +155,38 @@ export const IDIRCompanySearchResults = memo(
...
-      <div className="table-container idir-search-results">
-        <MaterialReactTable table={table} />
+      <div className="table-container idir-company-search-results">
+        {data?.items?.length !== 0 && <MaterialReactTable table={table} />}
+        {data?.items?.length === 0 &&
+          <>
+            <Stack style={{ width: "900px" }} spacing={2}>
+              <NoRecordsFound />
+              <Box
+                display="flex"
```
Comment (zgong-gov) [inline frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx]: You can add a classname of "create-company-btn" here, and move these styles into the scss file as above.

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss [hunk 3]
```diff
@@ -17,4 +18,13 @@
     padding: 0;
     max-width: 3rem;
```
Comment (zgong-gov) [inline frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:27]: Inside here (&#{&} which is equivalent to '.idir-company-search-results.idir-company-search-results'), you can add `.create-company-btn {
  display: flex;
  justify content: center;
  align-items: center;
  text-align: center;
}`

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx [hunk 4]
```diff
@@ -153,8 +155,38 @@ export const IDIRCompanySearchResults = memo(
...
+                justifyContent="center"
+                alignItems="center"
+                minHeight="30vh"
```
Comment (zgong-gov) [inline frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx]: Try not to use min-height, and see if paddings are needed (depending on the wireframe specs)

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss (hunk 3 above)
Comment (zgong-gov) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:27]: You could also add padding inside .create-company-btn if needed (depending on the wireframe specs), and remove min-height if possible.

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss [hunk 5]
```diff
@@ -17,4 +18,13 @@
...
   }
+
+  .create-company-img {
+    width: 6rem;
+  }
+
+  .button-outline {
+    border: orbcStyles.$bc-border-grey 1px solid;
```
Comment (krishnan-aot) [inline frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:35]: Doesn't it need a border radius?

Comment (krishnan-aot) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:35]: also hover state

Comment (erikataot) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:35]: yup added

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx (hunk 2 above)
Comment (erikataot) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx]: ok, I guess it's worth talking at some point about how much we hide in SCSS vs. using declarative properties in the UI widgets. Updated.

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss (hunk 3 above)
Comment (erikataot) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:27]: done.

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx (hunk 4 above)
Comment (erikataot) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.tsx]: yup thanks - double checked out the paddings

File: frontend/src/common/components/table/NoRecordsFound.tsx (hunk 1 above)
Comment (zgong-gov) [reply frontend/src/common/components/table/NoRecordsFound.tsx:10]: do you still have problems with loading this image? or can I resolve this now?

File: frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss (hunk 5 above)
Comment (erikataot) [reply frontend/src/features/idir/search/components/IDIRCompanySearchResults.scss:35]: updated the hover state.

**Review by zgong-gov — APPROVED** [review]

**Pull request merged**